### ISSUE TITLE
PersistentSession is never removed in MapDBSessionsStore.m_persistentSessions and never updated.

### ISSUE THREAD

**Issue by runafter** [moquette-io/moquette#145]
Hi, @andsel
I appreciate your works.

I update moquette to new version (0.8). But I found werid situation.

Steps to reproduce:
1. connect with clean session true
2. disconnect
3. connect with clean session false
4. subscribe a topic
5. disconnect

When disconnected, should not be invoked CleanSession.cleanSession() and should not delete subscriptions.
But it is, because ClientSession.cleanSession is never updated when connected.

```
public class ProtocolProcessor {
    ...

    public void processConnect(ServerChannel session, ConnectMessage msg) {
        ...

        ClientSession clientSession = m_sessionsStore.sessionForClient(msg.getClientID());
        boolean isSessionAlreadyStored = clientSession != null;
        if (!msg.isCleanSession() && isSessionAlreadyStored) {
            okResp.setSessionPresent(true);
        }
        session.write(okResp);
        m_interceptor.notifyClientConnected(msg);

        if (!isSessionAlreadyStored) {
            LOG.info("Create persistent session for clientID <{}>", msg.getClientID());
            clientSession = m_sessionsStore.createNewSession(msg.getClientID(), msg.isCleanSession());
        } else {
            LOG.info("Use old session for clientID <{}> with clean sesson {}", msg.getClientID(), clientSession.cleanSession);
        }
        clientSession.activate();
        if (msg.isCleanSession()) {
            clientSession.cleanSession();
        }
        LOG.info("Connected client ID <{}> with clean session {}", msg.getClientID(), msg.isCleanSession());
        if (!msg.isCleanSession()) {
            //force the republish of stored QoS1 and QoS2
            republishStoredInSession(clientSession);
        }
        LOG.info("CONNECT processed");
    }
}
```

In connection process, use clean session  value using msg.isCleanSession().
And Do not update clean session value to MapDBSessionsStore.m_persistentSessions

But in disconnection process, use ClientSession.cleanSession copied from MapDBSessionsStore.m_persistentSessions.

What should I do to fix it?

**Comment by andsel** [comment]
Hi @runafter thankyou for your bug report, I managed to solve it (storing it on persistence layer) 5c7e2b357f5c7f3a414982d078b2e22e64159b55.
Now it's fixed on 0.9 SNAPSHOT, could you test it again directly building from master?

 Andrea

**Comment by runafter** [comment]
Hi, @andsel .

I tested my test cases using master source code. And it works perfectly.

Thank you for your work.

**Comment by andsel** [comment]
Fixed in version 0.8.1
